File: project/jimmer-sql/src/test/java/org/babyfish/jimmer/sql/model/embedded/MachineDetail.java
package org.babyfish.jimmer.sql.model.embedded;

import org.babyfish.jimmer.sql.Column;
import org.babyfish.jimmer.sql.Embeddable;
import org.babyfish.jimmer.sql.Serialized;

import java.util.Map;

@Embeddable
public interface MachineDetail {

    @Column(name = "factory_map")
    @Serialized
    Map<String, String> factories();

    @Column(name = "patent_map")
    @Serialized
    Map<String, String> patents();
}

File: project/jimmer-sql-test/jimmer-sql-test-model-kotlin/src/main/kotlin/org/babyfish/jimmer/sql/kt/model/embedded/MachineDetail.kt
package org.babyfish.jimmer.sql.kt.model.embedded

import org.babyfish.jimmer.sql.Column
import org.babyfish.jimmer.sql.Embeddable
import org.babyfish.jimmer.sql.Serialized

@Embeddable
interface MachineDetail {

    @Column(name = "factory_map")
    @Serialized
    val factories: Map<String, String>

    @Column(name = "patent_map")
    @Serialized
    val patents: Map<String, String>
}